tests/unit/igraph_percolation.c

```diff
@@ -0,0 +1,198 @@
+/*
+    IGraph library.
+    Copyright (C) 2025  The igraph development team <igraph@igraph.org>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+#include "test_utilities.h"
+
+igraph_error_t percolate_b(igraph_t *graph, igraph_vector_int_t *edge_indices) {
+    igraph_vector_int_t outputs;
+    IGRAPH_VECTOR_INT_INIT_FINALLY(&outputs, 0);
+
+    IGRAPH_CHECK(igraph_bond_percolation(graph, &outputs, edge_indices));
+
+    print_vector_int(&outputs);
+
+    igraph_vector_int_destroy(&outputs);
+    IGRAPH_FINALLY_CLEAN(1);
+    return IGRAPH_SUCCESS;
+}
+
+igraph_error_t largest_component_b(igraph_t *graph, igraph_integer_t *size) {
+    igraph_vector_int_t outputs;
+    IGRAPH_VECTOR_INT_INIT_FINALLY(&outputs, 0);
+
+    IGRAPH_CHECK(igraph_bond_percolation(graph, &outputs, NULL));
+
+    *size = VECTOR(outputs)[igraph_vector_int_size(&outputs) -1];
+
+    igraph_vector_int_destroy(&outputs);
+    IGRAPH_FINALLY_CLEAN(1);
+    return IGRAPH_SUCCESS;
+
+}
+
+int test_bond(void) {
+    // test with normal graph and provided edge list
+
+    igraph_t k_3, c_4, karate, random;
+    igraph_integer_t size = 0;
+    igraph_full(&k_3, 3, false, false);
+
+    igraph_small(&c_4, 4, IGRAPH_UNDIRECTED, 0, 1, 1, 2, 2, 3, 3, 0, -1);
+
+    printf("K_3 graph percolation curve, no provided edge sequence:\n");
+    IGRAPH_CHECK(percolate_b(&k_3, NULL));
```

Comment:
`IGRAPH_CHECK` should only be used in functions that return `igraph_error_t`. Let's remove the error checks and make the function return `void`.

Note that when running tests, if an error occurs, the program aborts immediately and the test fails.

igraph supports "error handlers". The handler decides what to do upon error. The default handler prints the error and aborts. The handler used in Python will instead throw an appropriate Python exception and runs the function that calls all destructors on the "finally" stack.

Since the default behaviour is to abort, we don't need to do error handling in tests.